app/src/main/java/com/mapbox/maps/testapp/utils/StyleSwitcher.kt

```diff
@@ -0,0 +1,119 @@
+package com.mapbox.maps.testapp.utils
+
+import android.os.Handler
+import android.os.Looper
+import com.mapbox.bindgen.Value
+import com.mapbox.maps.*
+
+class StyleSwitcher(
+  val mapboxMap: MapboxMap,
+  initialStyle: Style
+) {
+  private lateinit var currentStyle: Style
+  private val mainHandler = Handler(Looper.getMainLooper())
+  private val originalProperties = mutableMapOf<Pair<String, String>, Value>()
+  private var styleTransitionDelayMs: Long = 0
+  private var onStyleSwitched: ((Style) -> Unit)? = null
+
+  init {
+    mapboxMap.subscribe(
+      object : Observer() {
+        override fun notify(event: Event) {
+          when (event.type) {
+            // when map loaded, fade-in layers using original values
+            MapEvents.MAP_LOADED -> {
+              for (originalProperty in originalProperties) {
+                currentStyle.setStyleLayerProperty(
+                  originalProperty.key.first,
+                  originalProperty.key.second + "-color-transition",
+                  Value(
+                    hashMapOf(
+                      "duration" to Value(styleTransitionDelayMs),
+                      "delay" to Value(0)
+                    )
+                  )
+                )
+                currentStyle.setStyleLayerProperty(
+                  originalProperty.key.first,
+                  originalProperty.key.second + "-color",
+                  originalProperty.value
+                )
+              }
+              originalProperties.clear()
+              onStyleSwitched?.invoke(currentStyle)
+            }
+          }
+        }
+      },
+      listOf(MapEvents.MAP_LOADED)
+    )
+    currentStyle = initialStyle
+  }
+
+  fun switchStyle(
```

Comment:
Could actually be an extension function on MapboxMap? Current style get be got through MapboxMap#getStyle API.

Reply:
Assuming it's not part of SDK but some util - I like current approach more (especially taking into account that this class has a valid ctor).

Reply:
yeah, for example activity I think it doesn't really matter, although we recommend users not keeping a reference of the style object, and always get it from the mapboxmap async. Do you think we can optimise it that way(remove the initial style and current style)?

Reply:
Done, thanks!